dvc/repo/metrics/diff.py

```diff
@@ -12,6 +12,13 @@ def diff(repo, *args, a_rev=None, b_rev=None, **kwargs):
     b_rev = b_rev or "workspace"
 
     metrics = repo.metrics.show(*args, **kwargs, revs=[a_rev, b_rev])
+
+    # workspace may have been replaced by active branch
+    workspace_rev = (a_rev == "workspace") or (b_rev == "workspace")
+    if workspace_rev and "workspace" not in metrics:
+        active_branch = repo.scm.active_branch()
+        metrics["workspace"] = metrics[active_branch]
```

Comment:
Seems that this could be reused in `repo/params/diff`, @dberenbaum wDYT of also including it in this P.R.

Reply:
Good point, added!